server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/CompositeAggregationBuilder.java

```diff
@@ -235,8 +236,12 @@ protected AggregatorFactory doBuild(
                 Object obj = after.get(sourceName);
                 if (configs[i].missingBucket() && obj == null) {
                     values[i] = null;
-                } else if (obj instanceof Comparable) {
-                    values[i] = (Comparable<?>) obj;
+                } else if (obj instanceof Comparable<?> c) {
+                    values[i] = c;
+                } else if (obj instanceof Map<?, ?> && configs[i].fieldType().getClass() == TimeSeriesIdFieldType.class) {
+                    // If input is a _tsid map, encode the map to the _tsid BytesRef
+                    // TODO: Must add validations
+                    values[i] = configs[i].format().parseBytesRef(obj);
```

Comment:
In the `after` part of the query, the `_tsid` field is passed as a `Map`.

Here we encode the Map to its bytesRef representation. From this point on, we treat the tsid `after` key as a binary value

Reply:
Another option could be to encode the json map in a string ? These values are not formatted for visualizations, they are part of the after key so the content an be obfuscated.

Reply:
We could base64 encode the tsid and slap it in the string. That'd obfuscate it plenty. I didn't like that idea before Christmas but I can't really remember why now....

Reply:
For things like search_after I think it's nice to have a readable after key. That we're formatting dates in composite makes me think we  want to format tsid - and the map *is* the formatted version. But if we just need *some* key - the base64 encoded version is easier.